### PR TITLE
DBZ-561 Provided option to flatten structs in MongoDB unwrapping SMT.

### PR DESCRIPTION
**Pull request by sairam881990**
https://issues.jboss.org/browse/DBZ-561

### CODE REVIEW

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java [hunk 1]
```diff
@@ -39,6 +42,8 @@
 
     private static final String SERVER_NAME = "serverX.";
     private static final String PREFIX = SERVER_NAME + ".";
+    private static final String FLATTEN_STRUCT = "flatten.struct";
+    private static final String DELIMITER = "delimiter";
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java]: I'd suggest to name it "flatten.delimiter" "flatten.struct.delimiter" to make its meaning clearer.

Comment (sairam881990) [reply debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java]: I will make these changes by the next commit.

**Comment by sairam881990** [comment]
I will reproduce the same and see what needs to be done in order to fix this.

**Comment by sairam881990** [comment]
@gunnarmorling I see that we actually need to build the document first from the update document. I mean suppose if the user does the following update 

db.customers.update(
   {
    _id : NumberLong("1005")
   },
   {
     $set : {
       "address.city" : 'Canberra'
     }     
   }
);

We would receive the patch as the field "address.city" which is getting built as the field address.city which is the issue. So we need to parse and build a document before building the schema and the struct. I have started looking into that and if you have any suggestions for how to build the document do let me know.

**Comment by gunnarmorling** [comment]
Right; you'd have to split the name at the dots (dots are no legal part of plain field names, so this should be safe) and then create (nested) documents for all parts apart from the last one.

**Comment by gunnarmorling** [comment]
Hey @sairam881990, any news on that one? As a heads-up, we're planning to do the 0.7.4 release mid next week. Would be cool to include this change, if we got something by then :)

**Comment by sairam881990** [comment]
@gunnarmorling  I want some suggestion for how to create the documents from the dot notation so that i can fix up the code and checkin the code by mid of next week.

**Comment by gunnarmorling** [comment]
@sairam881990 I'm not exactly sure what you're after. I think you'd have to do the following:

* split the field by dots, so "address.city" becomes ["address", "city"]
* Iterate through that array, creating documents for all but the last; so here you'd create a document for "address" and add the last element ("city") to it; for "address.country.name", you'd  create a document for "address", add to it a document for "country" and add to that "name".

But I'm not sure whether that's the advice you were after; All what counts in the end is that the correct structure is emitted :)

**Comment by gunnarmorling** [comment]
Hey @sairam881990, checking in on this one. Any progress? Or problems with it?

**Comment by sairam881990** [comment]
Hey @gunnarmorling, I will try to fix it and checkin the code by next week.

**Comment by jpechane** [comment]
@sairam881990 @gunnarmorling Guys, I tried to address latest comments, please let me now if your fine with it

**Comment by gunnarmorling** [comment]
Closing, as superseded by #531. Thanks, @sairam881990 and @jpechane.

**Pull request closed without merging**